src/coreclr/jit/lsraxarch.cpp

```diff
@@ -2152,6 +2168,26 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, int* pDstCou
             }
 #endif // TARGET_X86
 
+            case NI_X86Base_DivRem:
+            case NI_X86Base_X64_DivRem:
+            {
+                assert(numArgs == 3);
+                assert(dstCount == 2);
+                assert(isRMW);
+
+                // DIV implicitly put op1(lower) to EAX and op2(upper) to EDX
+                srcCount += BuildOperandUses(op1, RBM_EAX);
+                srcCount += BuildOperandUses(op2, RBM_EDX);
+                srcCount += op3->isContained() ? BuildOperandUses(op3) : BuildDelayFreeUses(op3, op1);
```

Comment:
Is this going to work as expected?

Normally the second operand to `BuildDelayFreeUses` is the rmw node. However, in this case we have two RMW nodes (op1 and op2) and so we should likely need to indicate that `op3` is delay free with regards to both `op1` and `op2`.

Reply:
I believe it remains `delayRegFree` for the entire treeNode. We reset `pendingDelayFree` when processing the next `treeNode`.

Reply:
Hmmm, @kunalspathak, there is special logic that looks at `node` (`op3`) compared to `rmwNode` (`op1`)

and a note specifically around:
```cpp
// Note: we don't handle multi-reg vars here. It's not clear that there are any cases
// where we'd encounter a multi-reg var in an RMW context.
assert(!rmwNode->AsLclVar()->IsMultiReg());
```

So my guess is we either don't have sufficient tests covering this case or we aren't correctly tracking the return value as multi-reg. There is also has the later logic that says `if (node == rmwNode)` then it might not set the delay free.

So, I think we need another overload of `BuildDelayFreeUses` that takes both `rmwNodes` and does the right general checks.